allennlp/nn/decoding/decoder_trainers/maximum_marginal_likelihood.py

```diff
@@ -65,7 +64,8 @@ def decode(self,
         batch_scores = self._group_scores_by_batch(finished_states)
         loss = 0
         for scores in batch_scores.values():  # we don't care about the batch index, just the scores
-            loss += -util.logsumexp(torch.cat(scores))
+            print("scores", scores)
+            loss += -util.logsumexp(torch.cat([tensor.view(-1) for tensor in scores]))
```

Comment:
I think doing a squeeze here is tighter - I was confused when I read this, as it isn't immediately obvious that it's a 1D tensor and that doing a view like this is the same.

Reply:
the actual issue here (which I didn't dig into too deeply) is that sometimes the score was a 0-tensor and sometimes it was a 1-tensor, and the `.view()` is ensuring it's a 1-tensor.